magma/lib/magma/attributes/identifier_attribute.rb

```diff
@@ -15,43 +15,36 @@ def after_magma_model_set
     class Validation < Magma::Validation::Attribute::BaseAttributeValidation
       def validate(value, &block)
 
-        # Some identifier attributes have regexp validations, so we call super() to run those validations first
-        super if @attribute.validation
-
         gnomon_mode = Magma::Flags::GNOMON_MODE
         flag_value = @flags[gnomon_mode[:name]]
 
         # We skip validation if we have the gnomon flag or if it's value is set to none
-        return if flag_value.nil? or flag_value == gnomon_mode[:none]
+        if flag_value.nil? or flag_value == gnomon_mode[:none]
+          # Some identifier attributes have additional validations, so we call super() to run those validations first
+          super if @attribute.validation
+          return
+        end
 
         begin
         # Identifier mode - we just care about the existence of the identifier in the identifier table
           if flag_value == gnomon_mode[:identifier]
-            get_identifier(value)
+            identifier = Magma::Gnomon::Identifier.where(project_name: @model.project_name.to_s, identifier: value).first
+            raise "The identifier '#{value}' has not been assigned in Gnomon." unless identifier
           # Pattern mode - We must make sure the identifier conforms to a grammar
           elsif flag_value == gnomon_mode[:pattern]
-            identifier = get_identifier(value)
             grammar = Magma::Gnomon::Grammar.for_project(@model.project_name.to_s)
             raise "Grammar not found, identifier is: '#{value}'" unless grammar
 
-            rule = grammar.parser.rules[identifier.rule]
-            raise "No such #{identifier.rule} for #{@model.project_name.to_s}" unless rule
+            identifier_model = grammar.model_name(value)
```

Comment:
this method seems to do some parsing/ unrolling, why not just `@model.name` here?

Reply:
good catch, I was looking at stale test case, and thought I needed to infer it